src/message_private.h

```diff
@@ -74,7 +74,12 @@ typedef struct {
 //
 
 typedef struct {
-    sys_mutex_t          lock;
+    sys_mutex_t          lock,
+                         producer_lock,                   // These locks prevent either side from activating
```

Comment:
For code consistency we avoid the multiple vars per type syntax in favor of single var per type declaration.  In other words:

```
// Instead of
sys_mutex_t a, b, c;
// we do
sys_mutex_t a;
sys_mutex_t b;
sys_mutex_t c;
```
The extra bytes do slow the executable down a bit but it's really not noticeable.